idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/ui/screen/mode/iconpack/newpack/ui/viewmodel/NewPackViewModel.kt

```diff
@@ -0,0 +1,161 @@
+            inputHandler.state.collect { inputFieldState ->
+                _state.updateState {
+                    if (this is PickedState) {
+                        copy(
+                            packEditState = packEditState.copy(inputFieldState = inputFieldState),
+                            nextAvailable = inputFieldState.noErrors(),
+                        )
+                    } else {
+                        this
+                    }
+                }
+            }
+        }
+    }
+
+    fun onAction(action: NewPackAction) {
+        when (action) {
+            is SelectDestinationFolder -> updateDestinationPath(action.path)
+            is SaveDestination -> {
+                saveDestination()
+                initDefaultPack()
+            }
+            is AddNestedPack -> inputHandler.addNestedPack()
+            is RemoveNestedPack -> inputHandler.removeNestedPack(action.nestedPack)
+            is PreviewPackObject -> previewIconPackObject()
+            is SavePack -> saveIconPack()
+        }
+    }
+
+    fun onValueChange(change: InputChange) = viewModelScope.launch {
+        inputHandler.handleInput(change)
+    }
+
+    private fun updateDestinationPath(path: Path) {
+        _state.updateState {
+            ChooseExportDirectoryState(
+                iconPackDestination = path.absolutePathString(),
+                predictedPackage = PackageExtractor.getFrom(path.invariantSeparatorsPathString).orEmpty(),
+                nextAvailable = true,
+            )
+        }
+    }
+
+    private fun saveDestination() {
+        val directoryState = currentState.castOrNull<ChooseExportDirectoryState>() ?: return
+
+        inMemorySettings.updateIconPackDestination(directoryState.iconPackDestination)
+    }
+
+    private fun initDefaultPack() {
+        _state.updateState {
+            PickedState(packEditState = PackEditState(inputFieldState = inputHandler.state.value))
+        }
+    }
+
+    private fun previewIconPackObject() = viewModelScope.launch {
+        val editState = currentState.castOrNull<PickedState>()?.packEditState ?: return@launch
+        val inputFieldState = editState.inputFieldState
+
+        val iconPackCode = IconPackGenerator.create(
+            config = IconPackGeneratorConfig(
+                packageName = inputFieldState.packageName.text,
+                iconPackName = inputFieldState.iconPackName.text,
+                subPacks = inputFieldState.nestedPacks.map { it.inputFieldState.text },
+            ),
+        ).content
+        _events.emit(PreviewIconPackObject(code = iconPackCode))
+    }
+
+    private fun saveIconPack() {
+        val packEditState = currentState.castOrNull<PickedState>()?.packEditState ?: return
+
+        viewModelScope.launch {
+            val fieldState = packEditState.inputFieldState
+            inMemorySettings.updatePackageName(fieldState.packageName.text)
+            inMemorySettings.updateIconPackName(fieldState.iconPackName.text)
```

Comment:
Dup with `IconPackCreationViewModel.saveSettings`

Reply:
Some of these VMs are unused, intended for keeping them?

Reply:
removed, thanks!